### DISCUSSION TITLE
Saving metadata back to files?

### DISCUSSION THREAD

**Discussion by Ibycus314** [advplyr/audiobookshelf#847] [Q&A]
Brand new to Audiobookshelf, so I very well might be missing something obvious - is it possible to save updated metadata back to the file formats that support it? 
Most of my Audiobooks are m4b format, so generally have pretty good metadata, but there are a number that would be nice to update. I can update within Audiobookshelf, but it doesn’t seem to save it back out to the file. For example, I have Foundation by Isaac Asimov. After the release of the series on Apple TV, the title was updated to “Foundation - Apple Series Tie in” which is stupid. I updated the title in Audiobookshelf, but when I dump the metadata from the file, the old title still shows up.

**Comment by advplyr** [comment]
I think you are looking for the "Embed Metadata" experimental feature. You must enable experimental features in config to access this.

![image](https://user-images.githubusercontent.com/67830747/179831350-466dc843-35b6-4d72-bb40-c162d75c2bcf.png)

**Reply by Ibycus314** [reply]
Thanks, looks like what I’m looking for. Do you know if there is any way to do this for all library items? Rather than one at a time.

**Reply by TheSylus** [reply]
Would really be great to get this as a one click solution. I like embed all the meta date and it would take ages to open every single book.

**Comment by Mihai-B** [comment]
I don't think writing metadata to epub files is currently supported. Does anyone know if there are any plans to support this?

**Reply by advplyr** [reply]
Most likely not natively but there are plans to add a plugin system where it could be possible for someone to build a plugin. That would also be a long ways out.

**Comment by Mihai-B** [comment]
A plugin would also be enough as long as it saves the metadata in the epub files. For me this is the only missing feature for this application.

**Comment by cerinawithasea1** [comment]
I use Mp3tag and after the embed many if my tags have been removed or remapped in a way that mp3tag can no longer see them
with use of these tools two so popular for audibooks, somebody must have run into this before (i hope)
